src/client/QXmppTransferManager.cpp

```diff
@@ -1361,8 +1361,6 @@ QXmppTransferJob *QXmppTransferManager::sendFile(const QString &jid, QIODevice *
         job->d->sid = sid;
     job->d->fileInfo = fileInfo;
     job->d->iodevice = device;
-    if (device)
```

Comment:
Probably a better way is to check the device parent and reparent it only if its parent is `nullptr`? Seems like a sane and expected behavior to me.